RockPaperScissors/RockPaperScissors.cs

```diff
@@ -6,20 +6,111 @@ class Program
     {
         public static void Main()
         {
-            Console.WriteLine("Enter hand 1:");
-            string hand1 = Console.ReadLine().ToLower();
-            Console.WriteLine("Enter hand 2:");
-            string hand2 = Console.ReadLine().ToLower();
-            Console.WriteLine(CompareHands(hand1, hand2));
-
-            // leave this command at the end so your program does not close automatically
-            Console.ReadLine();
+
+            // declare and initialize variables
+            bool play = true;
+            string playch;
+            string hand2 = "";
+            int hand1score = 0;
+            int hand2score = 0;
+            
+
+
+            // do while loop to be performed at least once, then game continuation depends on user choice            
+            do
+            {
+                Console.WriteLine("Enter hand 1: (Rock, Paper, or Scissors)");
+                // store user choice converted to lower case in the hand1 variable
+                string hand1 = Console.ReadLine().ToLower();
+                // declare and initialize new random variable to hold computer choice
+                Random rnd = new Random();
+                // create a string array of possible choices
+                string[] handchoices = {"rock", "paper", "scissors"};
+                // pick a random number between 0 and 2 inclusive and assign to variable
+                int choice_index = rnd.Next(0, 3);
+                // assign the randomly chosen index of the choice array to the variable hand2
+                hand2 = handchoices[choice_index];
+                // print the computer choice to screen
+                Console.WriteLine("Computer chooses {0}", hand2);
+                // call the CompareHands function while passing in hand1 and hand2 as parameters and output result to screen
+                Console.WriteLine(CompareHands(hand1, hand2, hand1score, hand2score));
+                //Console.WriteLine(KeepScore(hand1score, hand2score));
+
+                // ask user if wants to play again
+                Console.WriteLine("Would you like to play again? [y/n]:");
+                playch = Console.ReadLine();
+                // if user answers yes, then set play to true and loop back to the start of loop
+                if (playch == "y" || playch == "Y" || playch == "Yes" || playch == "YES" || playch == "yes")
+                    {
+                        play = true;
+                    }
+                    else
+                    {
+                        // user does not want to play again, so quit program
+                        play = false;
+                        break;
```

Comment:
both lines here are not needed. `break` will cause the while to stop and `play = false` will also cause the loop to stop. The difference is when each causes the do...while to terminate.